client/components/utils/refreshToken.ts

```diff
@@ -0,0 +1,14 @@
+export const refreshTokenSetup = (res) => {
+
+    let refreshTiming = (res.tokenObj.expires_in || 3600 - 5 * 60 ) * 1000
+
+    const refreshToken = async () => {
+        const newAuthRes = await res.reloadAuthResponse();
+        refreshTiming = (newAuthRes.expires_in || 3600 - 5 * 60) * 1000;
+
+        setTimeout(refreshToken, refreshTiming)
+        
+    };
+    
+    setTimeout(refreshToken, refreshTiming)
+};
```

Comment:
These 2 `setTimeout` calls are essentially the same as a `setInterval`. The only possible difference is the `expires_in`, which can be different on two reloads. Since this isn't a problem, I think `setInterval` is more suitable to use.
```suggestion
export const refreshTokenSetup = res => {
  const refreshTiming = (res.tokenObj.expires_in || 3600 - 5 * 60) * 1000;

  setInterval(res.reloadAuthResponse, refreshTiming);
};
```